src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,387 @@
+import React from 'react'
+import {Box} from '..'
+import {isResponsiveValue, useResponsiveValue, ResponsiveValue} from '../hooks/useResponsiveValue'
+import {SxProp, merge, BetterSystemStyleObject} from '../sx'
+import Heading from '../Heading'
+import {ArrowLeftIcon} from '@primer/octicons-react'
+import Link from '../Link'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
+import {get} from '../constants'
+const REGION_ORDER = {
+  ContextArea: 0,
+  TitleArea: 1,
+  Description: 2,
+  Navigation: 3,
+}
+
+// Types that are shared between sub components
+export type sharedPropTypes = {
+  hidden?: boolean | ResponsiveValue<boolean>
+} & SxProp
+
+// Default state for the `visible` prop when a sub component is only visible on narrow viewport
+const hiddenOnRegularAndWide = {
+  narrow: false,
+  regular: true,
+  wide: true,
+}
+
+// Default state for the `visible` prop when a sub component is visible on regular and wide viewport
+const hiddenOnNarrow = {
+  narrow: true,
+  regular: false,
+  wide: false,
+}
+
+// Root
+// -----------------------------------------------------------------------------
+export type PageHeaderProps = {
+  'aria-label'?: React.AriaAttributes['aria-label']
+  // TODO: check if we need any other types
+  as?: React.ElementType | 'header' | 'div'
+} & sharedPropTypes
+
+const Root: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({children, sx = {}, as = 'div'}) => {
```

Comment:
Oh great question! As far as I understand that wouldn't be valid in React 18 because they are removing implicit children type declaration in React.FC

Let me know if I missed anything! 

Resources might be helpful
-changelog: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/46691
- a blog post of one of the maintainers of DefinitelyTyped: https://solverfox.dev/writing/no-implicit-children/

Reply:
Oops - I forgot about that. Ignore what I said!